### PR TITLE
uhh this is makeembed.py, feel free to add changes or such.

### LINKED ISSUE #10: Add an embed command.
Add a command that generates embeds using either JSON parsing, direct python code, or an interactive setup. I think this should probably be locked to Cat Devs.

This would be useful because it would allow us to see how embeds look before we try committing them many times.

Anyone could implement this.

### PR DESCRIPTION
**Pull request by dannynotsmart**
Closes #10.

my first time contributing
also i probably forgot a bunch of things which is why YOU GUYS exist

### CODE REVIEW

**Review by Kronifer — COMMENTED** [review]
Did you lint and or use black?

**Comment by dannynotsmart** [comment]
what are these things

**Comment by Kronifer** [comment]
CONTRIBUTING.md

**Review by Objectivitix — CHANGES_REQUESTED** [review]
Pretty good, I suggest two little changes.

1. pull color data from config.yml, but if the issue is approved then soon you'll pull them from constants.py
2. down in the error part, you made the timestamp to be a datetime object. mind putting a strftime and putting that into human-readable format?

thanks!

**Comment by Objectivitix** [comment]
also, could you add docstrings and make your code comply with pep8?

**Comment by dannynotsmart** [comment]
BLACK WORKS YAYYYYY 
ok i gota add the color thing

**Comment by dannynotsmart** [comment]
big update

**Review by Objectivitix — APPROVED** [review]
alright, this looks super good! about the time formatting, it turns out the internal `__str__` dunder method does all that for you. approved!!

**Comment by dannynotsmart** [comment]
hehe

File: bot/exts/fun/makeembed.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
+import discord
+from discord.ext import commands
+import asyncio
+import datetime
+from bot.utilities import get_yaml_val
+""" import dem libs"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:6]: Why is there a docstring here it isnt needed

File: bot/exts/fun/makeembed.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+class makeembed(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+ 
+    @commands.command()
+    async def makeembed(self, ctx):
+
+        data = get_yaml_val("config.yml", "colors")
+        """ where all the colors can be found in a dict"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:17]: Unneeded docstring

File: bot/exts/fun/makeembed.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+""" import dem libs"""
+
+
+class makeembed(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+ 
+    @commands.command()
+    async def makeembed(self, ctx):
+
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:15]: unneeded whitespace

File: bot/exts/fun/makeembed.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        data = get_yaml_val("config.yml", "colors")
+        """ where all the colors can be found in a dict"""
+        questions = [
+            "What is the value of **title**?",
+            "What is the value of **description**?",
+            "What is the value of **footer**?",
+            "What is the value of **author**?",
+        ]
+        """ ultimate questionare"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:24]: Docstring

File: bot/exts/fun/makeembed.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        answers = []
+        """ where i save the answers"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:26]: Docstring

File: bot/exts/fun/makeembed.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        keys = list(data.keys())
+        """ converting the color keys in data to a list so i can reference it later """
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:28]: Docstring

File: bot/exts/fun/makeembed.py [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+        
+        def check(m):
+            return m.channel == ctx.channel and m.author == ctx.author
+        """ the check for self.bot.wait_For """
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:32]: Docstring

File: bot/exts/fun/makeembed.py [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+
+            msg = await self.bot.wait_for("message", check=check)
+            if msg.content == "cancel":
+                await ctx.send("Ending Process!")
+                return
+            else:
+                answers.append(msg.content)
+                await msg.add_reaction("✔️")
+                await asyncio.sleep(1)
+         """ asks the questions, waits for an response. if the msg is cancel the entire process ends """
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:45]: Docstring

File: bot/exts/fun/makeembed.py [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+                color = data[content.lower()]
+                await msg.add_reaction("✔️")
+                break
+
+            else:
+                await msg.add_reaction("❌")
+                await ctx.send("Not a valid color!")
+                await asyncio.sleep(1)
+                
+           """ asks the user for the color, checks if the message is a valid key in the data dict. if so, add the color to the answers. """
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:67]: Docstring

File: bot/exts/fun/makeembed.py [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+                await msg.add_reaction("❌")
+                await ctx.send("Not a valid color!")
+                await asyncio.sleep(1)
+                
+           """ asks the user for the color, checks if the message is a valid key in the data dict. if so, add the color to the answers. """                
+
+        embed = discord.Embed(title=answers[0], description=answers[1], color=color)
+        embed.set_footer(text=answers[2])
+        embed.set_author(name=answers[3], icon_url=ctx.author.avatar_url)
+        """ the embed we can make so far """
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:72]: Docstring

File: bot/exts/fun/makeembed.py [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+                    break
+
+                else:
+                    await ctx.send("Please answer with `y` or `n`")
+
+            else:
+                await ctx.send("You have no available fields left")
+                await asyncio.sleep(1)
+                break
+         """ basically asking for if user wants add_field. there is 5 available ones since idk how much to limit. pretty self explanatory but code looks ugly sry"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:117]: Docstring

File: bot/exts/fun/makeembed.py [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+        
+        
+        await ctx.send(
+            "Would you like an timestamp? Say `yes` or reply with something else for no"
+        )
+        msg = await self.bot.wait_for("message", check=check)
+        if msg.content == "yes":
+            embed.timestamp = datetime.datetime.utcnow()
+        """ asks for the timestamp cuz yeahhhh"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:126]: Docstring

File: bot/exts/fun/makeembed.py [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+                    new = await text.send(embed=embed)
+
+                    em = discord.Embed(description=f"[Sent!]({new.jump_url})")
+                    em.set_author(name=ctx.author, icon_url=ctx.author.avatar_url)
+                    em.timestamp = datetime.datetime.utcnow()
+                    await ctx.send(embed=em)
+
+            else:
+                await ctx.send("ok bye")
+         """ asks if they want to send the made embed into another channel. i think this is good if we want like fancy announcements or sum but this is only restricted to admins"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:154]: Docstring

File: bot/exts/fun/makeembed.py [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+    @makeembed.error
+    async def makeembed_error(self, ctx, error):
+        await ctx.send("An Error occured")
+        error_report = discord.Embed(
+            description=f"```{error}```",
+            color=0xFF0000,
+            timestamp=datetime.datetime.utcnow(),
+        )
+        await ctx.send(embed=error_report)
+     """ error handelr??"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:166]: Docstring

File: bot/exts/fun/makeembed.py [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+
+def setup(bot):
+    bot.add_cog(makeembed(bot))
+
+ """ sincerely, cat"""
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:171]: Docstring

**Review by Kronifer — CHANGES_REQUESTED** [review]
If a docstring is unneeded, remove it. Otherwise, change them to comments using the # symbol without quotes

**Comment by dannynotsmart** [comment]
ok i did the changes

File: bot/exts/fun/makeembed.py [hunk 16]
```diff
@@ -0,0 +1,169 @@
...
+            "What is the value of **author**?",
+        ]
+
+        answers = []
+
+        keys = list(data.keys())
+        
+        def check(m):
+            return m.channel == ctx.channel and m.author == ctx.author
+        """ the check for self.bot.wait_For """
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:30]: Still docstring

File: bot/exts/fun/makeembed.py [hunk 17]
```diff
@@ -0,0 +1,169 @@
...
+            msg = await self.bot.wait_for("message", check=check)
+            
+            if msg.content == "cancel":
+                await ctx.send("Ending Process!")
+                return
+            
+            answers.append(msg.content)
+            await msg.add_reaction("✔️")
+            await asyncio.sleep(1)
+         """ asks the questions, waits for an response. if the msg is cancel the entire process ends """
```
Comment (Kronifer) [inline bot/exts/fun/makeembed.py:43]: Still docstring

**Review by Kronifer — APPROVED** [review]
lgtm, merged

**Pull request merged**